aspnetcore/tutorials/web-api-vsc.md

```diff
@@ -0,0 +1,272 @@
+|DELETE /api/todo/{id}  &nbsp;  &nbsp; | Delete an item &nbsp;  &nbsp;  | None. No request body-  | None|
+     
+<br>     
+    
+The following diagram shows the basic design of the app.
+
+![The client is represented by a box on the left and submits a request and receives a response from the application, a box drawn on the right. Within the application box, three boxes represent the controller, the model, and the data access layer. The request comes into the application's controller, and read/write operations occur between the controller and the data access layer. The model is serialized and returned to the client in the response.](first-web-api/_static/architecture.png)
+
+* The client is whatever consumes the web API (browser, mobile app, and so forth). We aren’t writing a client in this tutorial. We'll use [Postman](https://www.getpostman.com/) to test the app.
+
+* A *model* is an object that represents the data in your application. In this case, the only model is a to-do item. Models are represented as simple C# classes (POCOs).
+
+* A *controller* is an object that handles HTTP requests and creates the HTTP response. This app will have a single controller.
+
+* To keep the tutorial simple, the app doesn’t use a persistent database. Instead, it stores to-do items in an in-memory database. 
+
+### Set up your development environment
+
+Download and install [.NET Core](https://microsoft.com/net/core) and [Visual Studio Code](https://code.visualstudio.com) with the [C# extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode.csharp).
+
+### Create the project
+
+From a console, run the following commands:
+
+```console
+mkdir TodoApi
+cd TodoApi
+dotnet new webapi
+```
+
+Open the *TodoApi* folder in Visual Studio Code (VS Code). 
+
+- Select **Yes** to the **Warn** message "Required assets to build and debug are missing from 'TodoApi'. Add them?"
+- Select **Restore** to the **Info** message "There are unsresolved dependencies".
+
+![VS Code with Warn Required assets to build and debug are missing from 'TodoApi'. Add them? Don't ask Again, Not Now, Yes and also Info - there are unresolved dependencies  - Restore - Close](web-api-vsc/_static/vsc_restore.png)
+
+Press ^F5 to build and run the program. In a browser naviage to http://localhost:5000/api/values . The following is displayed:
+
+`["value1","value2"]`
+
+### Add support for Entity Framework Core
+
+Edit the *TodoApi.csproj* file to install the [Entity Framework Core InMemory](https://docs.microsoft.com/en-us/ef/core/providers/in-memory/) database provider. This database provider allows Entity Framework Core to be used with an in-memory database.
+
+[!code-xml[Main](web-api-vsc/sample/TodoApi/TodoApi.csproj?highlight=13-15)]
+
+Run `dotnet restore` to download and install the EF Core InMemory DB provider. You can run `dotnet restore` from the terminal or enter `⌘⇧P` or `Ctrl+Shift+P` in VS Code and then type **.NET**. Select **.NET: Restore Packages**.
```

Comment:
I've decided to not give OS instructions like how to open a terminal. That's easy to google and Mac folks know how to do that. I'm removing ^F.